File: SDKs/Aspose.Words_Cloud_SDK_For_Ruby/lib/aspose_words_cloud/models/document.rb
module AsposeWordsCloud
  # 
  class Document < BaseObject
    attr_accessor :links, :file_name, :source_format, :is_encrypted, :is_signed, :document_properties
    # attribute mapping from ruby-style variable name to JSON key
    def self.attribute_map
      {
        
        # 
        :'links' => :'Links',
        
        # 
        :'file_name' => :'FileName',
        
        # 
        :'source_format' => :'SourceFormat',
        
        # 
        :'is_encrypted' => :'IsEncrypted',
        
        # 
        :'is_signed' => :'IsSigned',
        
        # 
        :'document_properties' => :'DocumentProperties'
        
      }
    end

    # attribute type
    def self.swagger_types
      {
        :'links' => :'Array<Link>',
        :'file_name' => :'String',
        :'source_format' => :'String',
        :'is_encrypted' => :'BOOLEAN',
        :'is_signed' => :'BOOLEAN',
        :'document_properties' => :'DocumentProperties'
        
      }
    end

    def initialize(attributes = {})
      return if !attributes.is_a?(Hash) || attributes.empty?

      # convert string to symbol for hash key
      attributes = attributes.inject({}){|memo,(k,v)| memo[k.to_sym] = v; memo}

      
      if attributes[:'Links']
        if (value = attributes[:'Links']).is_a?(Array)
          self.links = value
        end
      end
      
      if attributes[:'FileName']
        self.file_name = attributes[:'FileName']
      end
      
      if attributes[:'SourceFormat']
        self.source_format = attributes[:'SourceFormat']
      end
      
      if attributes[:'IsEncrypted']
        self.is_encrypted = attributes[:'IsEncrypted']
      end
      
      if attributes[:'IsSigned']
        self.is_signed = attributes[:'IsSigned']
      end
      
      if attributes[:'DocumentProperties']
        self.document_properties = attributes[:'DocumentProperties']
      end
      
    end

    def source_format=(source_format)
      allowed_values = ["Unknown", "Doc", "Dot", "DocPreWord97", "Docx", "Docm", "Dotx", "Dotm", "FlatOpc", "Rtf", "WordML", "Html", "Mhtml", "Epub", "Text", "Odt", "Ott", "Pdf", "Xps", "Swf", "Tiff", "Svg"]
      if source_format && source_format.to_i >= allowed_values.size
        fail "invalid value for 'source_format', must be one of #{allowed_values}"
      end
      @source_format = allowed_values[source_format.to_i]
    end

  end
end

File: SDKs/Aspose.Words-Cloud-SDK-for-.NET/src/Com/Aspose/Words/Model/Document.cs
using System;
using System.Text;
using System.Collections;
using System.Collections.Generic;
using System.Runtime.Serialization;
using Newtonsoft.Json;
using Newtonsoft.Json.Converters;

namespace Com.Aspose.Words.Model 
{
  /// <summary>
  /// Represents Words document DTO.
  /// </summary>  
  public class Document 
  {                       
        /// <summary>
        /// Gets the original format of the document.
        /// </summary>
        /// <value>Gets the original format of the document.</value>
        [JsonConverter(typeof(StringEnumConverter))]
        public enum SourceFormatEnum
        {
            
            /// <summary>
            /// Enum Unknown for "Unknown"
            /// </summary>            
            Unknown,
            
            /// <summary>
            /// Enum Doc for "Doc"
            /// </summary>            
            Doc,
            
            /// <summary>
            /// Enum Dot for "Dot"
            /// </summary>            
            Dot,
            
            /// <summary>
            /// Enum DocPreWord60 for "DocPreWord60"
            /// </summary>            
            DocPreWord60,
            
            /// <summary>
            /// Enum Docx for "Docx"
            /// </summary>            
            Docx,
            
            /// <summary>
            /// Enum Docm for "Docm"
            /// </summary>            
            Docm,
            
            /// <summary>
            /// Enum Dotx for "Dotx"
            /// </summary>            
            Dotx,
            
            /// <summary>
            /// Enum Dotm for "Dotm"
            /// </summary>            
            Dotm,
            
            /// <summary>
            /// Enum FlatOpc for "FlatOpc"
            /// </summary>            
            FlatOpc,
            
            /// <summary>
            /// Enum Rtf for "Rtf"
            /// </summary>            
            Rtf,
            
            /// <summary>
            /// Enum WordML for "WordML"
            /// </summary>            
            WordML,
            
            /// <summary>
            /// Enum Html for "Html"
            /// </summary>            
            Html,
            
            /// <summary>
            /// Enum Mhtml for "Mhtml"
            /// </summary>            
            Mhtml,
            
            /// <summary>
            /// Enum Epub for "Epub"
            /// </summary>            
            Epub,
            
            /// <summary>
            /// Enum Text for "Text"
            /// </summary>            
            Text,
            
            /// <summary>
            /// Enum Odt for "Odt"
            /// </summary>            
            Odt,
            
            /// <summary>
            /// Enum Ott for "Ott"
            /// </summary>            
            Ott,
            
            /// <summary>
            /// Enum Pdf for "Pdf"
            /// </summary>            
            Pdf,
            
            /// <summary>
            /// Enum Xps for "Xps"
            /// </summary>            
            Xps,
            
            /// <summary>
            /// Enum Swf for "Swf"
            /// </summary>            
            Swf,
            
            /// <summary>
            /// Enum Tiff for "Tiff"
            /// </summary>            
            Tiff,
            
            /// <summary>
            /// Enum Svg for "Svg"
            /// </summary>            
            Svg
        }

        /// <summary>
        /// Gets the original format of the document.
        /// </summary>
        /// <value>Gets the original format of the document.</value>
        public SourceFormatEnum? SourceFormat { get; set; }

        /// <summary>
        /// A list of links that originate from this document.
        /// </summary>
        /// <value>A list of links that originate from this document.</value>    
        public List<Link> Links { get; set; }

        /// <summary>
        /// Gets the name of the file.
        /// </summary>
        /// <value>Gets the name of the file.</value>    
        public string FileName { get; set; }

        /// <summary>
        /// Returns true if the document is encrypted and requires a password to open.
        /// </summary>
        /// <value>Returns true if the document is encrypted and requires a password to open.</value>    
        public bool? IsEncrypted { get; set; }

        /// <summary>
        /// Returns true if the document contains a digital signature. This property merely informs that a   digital signature is present on a document, but it does not specify whether the signature is valid or not.
        /// </summary>
        /// <value>Returns true if the document contains a digital signature. This property merely informs that a   digital signature is present on a document, but it does not specify whether the signature is valid or not.</value>    
        public bool? IsSigned { get; set; }

        /// <summary>
        /// Returns document properties.
        /// </summary>
        /// <value>Returns document properties.</value>    
        public DocumentProperties DocumentProperties { get; set; }

        /// <summary>
        /// Get the string presentation of the object
        /// </summary>
        /// <returns>String presentation of the object</returns>
        public override string ToString()  
        {
          var sb = new StringBuilder();
          sb.Append("class Document {\n");
          sb.Append("  Links: ").Append(Links).Append("\n");
          sb.Append("  FileName: ").Append(FileName).Append("\n");
          sb.Append("  SourceFormat: ").Append(SourceFormat).Append("\n");
          sb.Append("  IsEncrypted: ").Append(IsEncrypted).Append("\n");
          sb.Append("  IsSigned: ").Append(IsSigned).Append("\n");
          sb.Append("  DocumentProperties: ").Append(DocumentProperties).Append("\n");
          sb.Append("}\n");
          return sb.ToString();
        }
    }
}
